gdx/src/com/badlogic/gdx/graphics/g2d/DistanceFieldBitmapFont.java

```diff
@@ -0,0 +1,192 @@
+ * NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.badlogic.gdx.graphics.g2d;
+
+import com.badlogic.gdx.files.FileHandle;
+import com.badlogic.gdx.graphics.Texture.TextureFilter;
+import com.badlogic.gdx.graphics.glutils.ShaderProgram;
+import com.badlogic.gdx.utils.Array;
+
+/** Renders bitmap fonts using distance field textures, @see <a href="https://github.com/libgdx/libgdx/wiki/Distance-field-fonts">Distance Field Fonts Wiki Article</a>
+ * For usage init the SpriteBatch with the {@link #getDistanceFieldShader()} shader
+ * Attention: Breaks batching!
+ * @author Florian Falkner */
+public class DistanceFieldBitmapFont extends BitmapFont {
+	private static final String SMOOTHING_UNIFORM = "u_smoothing";
+	private float distanceFieldSmoothing;
+	private static ShaderProgram shader;
+	
+	public DistanceFieldBitmapFont (BitmapFontData data, Array<TextureRegion> pageRegions, boolean integer) {
+		super(data, pageRegions, integer);
+	}
+
+	public DistanceFieldBitmapFont (BitmapFontData data, TextureRegion region, boolean integer) {
+		super(data, region, integer);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, boolean flip) {
+		super(fontFile, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, FileHandle imageFile, boolean flip, boolean integer) {
+		super(fontFile, imageFile, flip, integer);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, FileHandle imageFile, boolean flip) {
+		super(fontFile, imageFile, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, TextureRegion region, boolean flip) {
+		super(fontFile, region, flip);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile, TextureRegion region) {
+		super(fontFile, region);
+	}
+
+	public DistanceFieldBitmapFont (FileHandle fontFile) {
+		super(fontFile);
+	}
+
+	@Override
+	public BitmapFontCache newFontCache () {
+		for (int i = 0; i < regions.size; i++) {
+			getRegion(i).getTexture().setFilter(TextureFilter.Linear, TextureFilter.Linear);
+		}
+		
+		return new DistanceFieldFontCache(this, integer);
+	}
+	
+	/**
+	 * @return The distance field smoothing factor for this font. */
+	public float getDistanceFieldSmoothing () {
+		return distanceFieldSmoothing;
+	}
+
+	/**
+	 * @param distanceFieldSmoothing Set the distance field smoothing factor for this font. */
+	public void setDistanceFieldSmoothing (float distanceFieldSmoothing) {
+		this.distanceFieldSmoothing = distanceFieldSmoothing;
+	}
+	
+	/**
+	 * @return ShaderProgram Returns a singleton instance for the distance field shader, SpriteBatch needs to have this shader set for rendering distance field fonts. */
+	public static ShaderProgram getDistanceFieldShader() {
+		if (shader == null) {
+			shader = createDistanceFieldShader();
+			shader.setUniformf(SMOOTHING_UNIFORM, 0);
+		}
+		
+		return shader;
```

Comment:
Regarding passing the shader: sorry, I don't get it. 
The shader is only needed in the SpriteBatch, but there's no getShader() method in SpriteBatch, so DistanceFieldFontCache can't access it.
In which constructor should the shader be passed? DistanceFieldBitmapFont, IMHO not really the right place?